File: examples/factorial.js
eval(`function mult(a, b, c) {
    if(c > 1){                        
        return mult(a, b + a, c - 1)
    }else{
        return b
    }    
}
function factorial(n){    
    if (n == 1) {
        return n
    }else{               
        return mult(n, n, factorial(n - 1))        
    }
}`)
let t = Date.now()
let res = factorial(10)
console.log(res == 3628800 ?  "OK: " + res : "WRONG: " + res, "time:", Date.now()-t)

File: examples/factorial.py
import datetime, time
import sys
sys.setrecursionlimit(10**6)

def mult(a, b, c):
    if c > 1:                      
        return mult(a, b + a, c - 1)
    else:
        return b

def factorial(n):    
    if n == 1:
        return n
    else:               
        return mult(n, n, factorial(n - 1))       


t = datetime.datetime.now()
res = factorial(10)
print("OK: " + str(res) if res == 3628800 else "WRONG: " + str(res))
print("time:", (datetime.datetime.now()-t))